plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseReplicationModule.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jkiss.dbeaver.ext.altibase.model;
+
+/*
+ * An abstract class for Replication sender and receiver.
+ */
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+
+import java.sql.SQLException;
+
+public abstract class AltibaseReplicationModule extends AltibaseObject<AltibaseReplication> {
+
+    protected AltibaseReplication parent;
+    
+    protected AltibaseReplicationModule(AltibaseReplication parent) {
+        super(parent, parent.getName(), false);
+        this.parent = parent;
+    }
+    
+    @Override
+    public AltibaseReplication getParentObject() {
+        return parent;
+    }
+
+    @Override
+    public AltibaseDataSource getDataSource() {
+        return parent.getDataSource();
+    }
+
+    @Override
+    public String getDescription() {
+        return null;
+    }
+
+    @Override
+    public boolean isPersisted() {
+        return false;
+    }
+    
+    protected boolean hasData(JDBCResultSet resultSet) {
+        try {
+            return (resultSet.getRow() > 0);
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
Please add field 
`private static final Log log = Log.getLog(AltibaseReplicationModule.class);`
and use logger here
`log.debug(e);`

Reply:
> Please add field `private static final Log log = Log.getLog(AltibaseReplicationModule.class);` and use logger here `log.debug(e);`

This method is not used any more. So, I remove it. I'll keep your tip :)